MGMT-3077 Add support for NTP configuration

jtomasek opened the pull request:
- Add NTP source input field in advanced network configuration
- Add support for ntp-synced backend network validation
- Add NTP status to host detail

Depends on https://github.com/openshift/assisted-service/pull/675

Failing validation + NTP status unreachable:
![Screenshot from 2020-11-24 14-11-27](https://user-images.githubusercontent.com/1121740/100100237-c7472a00-2e60-11eb-8d29-b73e92a6d0ed.png)

Configuring NTP source:
![Screenshot from 2020-11-24 14-12-18](https://user-images.githubusercontent.com/1121740/100100252-d037fb80-2e60-11eb-9639-109300297186.png)

NTP synced:
![Screenshot from 2020-11-24 13-58-42](https://user-images.githubusercontent.com/1121740/100100276-db8b2700-2e60-11eb-90ab-e9908820f346.png)

mareklibra commented on src/components/hosts/HostRowDetail.tsx:
```diff
@@ -146,6 +149,10 @@ export const HostDetail: React.FC<HostDetailProps> = ({ inventory, host }) => {
         <DetailItem title="CPU architecture" value={inventory.cpu?.architecture || DASH} />
         <DetailItem title="CPU model name" value={inventory.cpu?.modelName || DASH} />
         <DetailItem title="CPU cores and clock speed" value={rowInfo.cpuSpeed} />
+        <DetailItem
```
Can we move this to the third column?
This one is under redesign in other open PRs.

jtomasek replied:
Done

nirfarkas commented:
I think what's missing in the notification is a way to fix this issue. Maybe something like: "Please specify an NTP server in the Advanced Networking section".
Also, does the NTP field always shown? if so, it should be mandatory and auto-populated (when everything is in sync, and empty for input if there's a problem.

mareklibra commented:
How is NTP handled in case of Day 2?

jtomasek commented:
> I think what's missing in the notification is a way to fix this issue. Maybe something like: "Please specify an NTP server in the Advanced Networking section".

I can add `notificationHints` constant to allow adding client side messages to failing validations. Currently only the backend provides the message.

> Also, does the NTP field always shown? if so, it should be mandatory and auto-populated (when everything is in sync, and empty for input if there's a problem.

Yes, the field is always shown. The field is by default pre-populated with `0.rhel.pool.ntp.org` and if the validation fails, user can provide different value. In general, user can fix the validation either by configuring the ntp sources manually on the host or configure ntp source via the UI.
I don't quite understand why the field should have additional logic then just enabling user to configure one ntp source for all hosts.

jtomasek commented:
> How is NTP handled in case of Day 2?

Technically it is still possible to make update_cluster API calls, however we don't do any operation like this on day 2 yet so we'd have to introduce a dialog to set this which may be quite confusing IMO. So for day 2 I'd prefer to notify user and let them manually fix the problem and ensure that host is ntp synced.

@YuviGold

jtomasek commented:
Updated the `ntpSource` field label and description, added client side hint to `ntp-synced` validation failure.

<img width="1113" alt="Screenshot 2020-11-25 at 15 56 16" src="https://user-images.githubusercontent.com/1121740/100244705-8bcb5f00-2f37-11eb-9e4a-6f39060d3505.png">

<img width="934" alt="Screenshot 2020-11-25 at 16 01 35" src="https://user-images.githubusercontent.com/1121740/100244714-8e2db900-2f37-11eb-96cf-fa76c205b490.png">

nirfarkas commented:
Hmmm... shouldn't the field be mandatory? I think it can't be empty.

jtomasek commented:
It can be empty. In some AI setups it has an empty default value.

nirfarkas commented:
Thanks, Jiri!
